### ISSUE TITLE
OpenCOR crashes on Ubuntu 16.04 when opening model

### ISSUE THREAD

**Issue by dbrnz** [opencor/opencor#1205]
This is with the latest sources from master (commit e98979a). 

[OpenCOR_Ubuntu_crash.zip](https://github.com/opencor/opencor/files/747385/OpenCOR_Ubuntu_crash.zip)

**Comment by dbrnz** [comment]
I can confirm that things work fine with the latest snapshot. The crash is with a clean build using commit e98979a; resetting OpenCOR's settings makes no difference.

**Comment by dbrnz** [comment]
Have just tried rebuilding with a fresh clone of the sources from GitHub and get the same crash. It occurs with any CellML file.

**Comment by agarny** [comment]
This is really weird. That commit (e98979af32cf4b5196a74cd13ddd03dfd35b2219) is the one on which the latest snapshot was generated (using a clean build). Note, however, that my Linux work is done using Ubuntu 14.04. I 'only' test on Ubuntu 16.04 afterwards. Anyway, I am going to try on Ubuntu 16.04 using a fresh clone and see how it goes...

**Comment by agarny** [comment]
Ok, I can now confirm the issue you reported...

As I mentioned above, I am building OpenCOR on 14.04 using GCC/G++ 4.9 (see https://github.com/opencor/opencor/commit/2aa154ea8e77fb2d9f75dd5c5de200c0a06f5406 for the reason). The CellML API binaries are also built that way, and they rely on libstdc++, which is different for GCC/G++ 4.9 and the version of GCC/G++ that is available on 16.04. This is one of the reasons I am shipping libstdc++ with the Linux version of OpenCOR, and this is the reason everything works fine with the 2017-02-01 snapshot.

In other words, I have just rebuilt the CellML API on 16.04 and used it with the OpenCOR version that I built on 16.04, and everything works fine.

So, I guess you have only recently moved to 16.04? I mean, I have been using GCC/G++ 4.9 on 14.04 for more than 7 months now...

I don't know whether we should drop 14.04 and just focus on 16.04, not least since Travis CI is (still!) using 12.04, but maybe we could still get things to build and our tests pass even if we were to focus only on 16.04.

So... views? Support 14.04 and later with the caveat you have come across for people who want to build OpenCOR on 16.04 rather than 14.04, or support 16.04 and later? @nickerso?
